docarray/array/stacked/array_stacked.py

```diff
@@ -70,205 +67,67 @@ class DocumentArrayStacked(AnyDocumentArray[T_doc]):
     """
 
     document_type: Type[BaseDocument] = AnyDocument
-    _docs: DocumentArray
+    _storage: ColumnStorage
 
     def __init__(
         self: T,
-        docs: Optional[Union[DocumentArray, Iterable[T_doc]]] = None,
+        docs: Sequence[T_doc],
         tensor_type: Type['AbstractTensor'] = NdArray,
     ):
-        self._doc_columns: Dict[str, 'DocumentArrayStacked'] = {}
-        self._tensor_columns: Dict[str, AbstractTensor] = {}
         self.tensor_type = tensor_type
-
-        self.from_iterable_document(docs)
-
-    def from_iterable_document(
-        self: T, docs: Optional[Union[DocumentArray, Iterable[T_doc]]]
-    ):
-        self._docs = (
-            docs
-            if isinstance(docs, DocumentArray)
-            else DocumentArray.__class_getitem__(self.document_type)(
-                docs, tensor_type=self.tensor_type
-            )
-        )
-        self.tensor_type = self._docs.tensor_type
-        self._doc_columns, self._tensor_columns = self._create_columns(
-            self._docs, tensor_type=self.tensor_type
+        self._storage = ColumnStorage.from_docs(
+            docs, document_type=self.document_type, tensor_type=self.tensor_type
         )
 
     @classmethod
-    def _from_da_and_columns(
+    def from_columns_storage(cls: Type[T], storage: ColumnStorage) -> T:
+        """
+        Create a DocumentArrayStack directly from a storage object
+        :param storage: the underlying storage.
+        :return: a DocumentArrayStack
+        """
+        da = cls.__new__(cls)
+        da.tensor_type = storage.tensor_type
+        da._storage = storage
+        return da
+
+    @classmethod
+    def validate(
```

Comment:
What is this used for? Isn't this usually a pydantic thing?

Reply:
Yes but we need it actually if I want to do 
```python
class MyDoc(BaseDocument):
   images: DocumetArrayStacked[Image]
```

Reply:
Ok. So we copy the pydantic validation interface, but acutally just call this method manually, right?